src/clj_sqs_extended/aws/sqs.clj

```diff
@@ -30,35 +30,51 @@
         builder (if creds (.withCredentials builder creds) builder)]
     (AmazonSQSExtendedClient. (.build builder) sqs-config)))
 
+(defn build-create-queue-request-with-attributes
+  [name
+   {:keys [fifo
+           visibility-timeout-in-seconds
+           kms-master-key-id
+           kms-data-key-reuse-period]}]
+  (cond-> (CreateQueueRequest. name)
+    fifo
+    (.addAttributesEntry "FifoQueue"
+                         "true")
+
+    visibility-timeout-in-seconds
+    (.addAttributesEntry "VisibilityTimeout"
+                         (str visibility-timeout-in-seconds))
+
+    kms-master-key-id
+    (.addAttributesEntry "KmsMasterKeyId"
+                         (str kms-master-key-id))
+
+    kms-data-key-reuse-period
+    (.addAttributesEntry "KmsDataKeyReusePeriodSeconds"
+                         (str kms-data-key-reuse-period))))
+
 (defn- create-queue
   ([sqs-client name]
    (create-queue sqs-client name {}))
 
   ([sqs-client name
     {:keys [fifo
+            visibility-timeout-in-seconds
             kms-master-key-id
-            kms-data-key-reuse-period]}]
-   (let [request (cond-> (CreateQueueRequest. name)
-                   fifo
-                   (doto (.addAttributesEntry
-                           "FifoQueue" "true"))
-                   kms-master-key-id
-                   (.addAttributesEntry
-                     "KmsMasterKeyId" kms-master-key-id)
-
-                   kms-data-key-reuse-period
-                   (.addAttributesEntry
-                     "KmsDataKeyReusePeriodSeconds" kms-data-key-reuse-period))]
-     (->> (.createQueue sqs-client request)
-          (.getQueueUrl)))))
+            kms-data-key-reuse-period]
```

Comment:
don't need to destructure the keys here since they're not used and this is a private fn